tools/unicode_properties_parse/format_width_estimate_intervals.cpp

```diff
@@ -0,0 +1,204 @@
+    static constexpr uint32_t max_u = 0x10'ffff;
+    std::vector<width_u> table;
+
+public:
+    table_u() : table(max_u + 1, width_u::is_1) {}
+    void fill_range(const range_u rng, const width_u width) {
+        const auto [from, to] = rng;
+        verify(from <= to && to <= max_u, impl_assertion_failed);
+        for (uint32_t u = from; u <= to; u++) {
+            table[u] = width;
+        }
+    }
+
+    void print_intervals() const {
+        // Print table for `_Width_estimate_intervals`.
+        using namespace std;
+        int c        = 0;
+        width_u last = table[0];
+        for (uint32_t u = 0; u <= max_u; u++) {
+            if (table[u] != last) {
+                cout << "0x" << hex << uppercase << u << "u, ";
+                if (++c == 12) {
+                    c = 0;
+                    cout << endl;
+                }
+            }
+            last = table[u];
+        }
+        cout << endl;
+    }
+
+    void print_clusters_1_vs_2(const table_u& other) const {
+        using namespace std;
+        vector<bool> cluster_table(max_u + 1, false);
+        for (uint32_t u = 0; u <= max_u; u++) {
+            if (table[u] == width_u::is_1 && other.table[u] == width_u::is_2) {
+                cluster_table[u] = true;
+            }
+        }
+
+        for (uint32_t u = 0; u <= max_u; u++) {
+            if (cluster_table[u]) {
+                uint32_t from = u;
+                uint32_t to   = from;
+                while (to + 1 <= max_u && cluster_table[to + 1]) {
+                    ++to;
+                }
+                if (from == to) {
+                    cout << hex << uppercase << "U+" << from << endl;
+                } else {
+                    cout << hex << uppercase << "U+" << from << "..U+" << to << endl;
+                }
+                u = to;
+            }
+        }
+    }
+};
+
+table_u get_table_cpp20() {
+    const range_u std_wide_ranges_cpp20[]{
+        {0x1100, 0x115F},
+        {0x2329, 0x232A},
+        {0x2E80, 0x303E},
+        {0x3040, 0xA4CF},
+        {0xAC00, 0xD7A3},
+        {0xF900, 0xFAFF},
+        {0xFE10, 0xFE19},
+        {0xFE30, 0xFE6F},
+        {0xFF00, 0xFF60},
+        {0xFFE0, 0xFFE6},
+        {0x1F300, 0x1F64F},
+        {0x1F900, 0x1F9FF},
+        {0x20000, 0x2FFFD},
+        {0x30000, 0x3FFFD},
+    };
+
+    table_u table;
+    for (const range_u rng : std_wide_ranges_cpp20) {
+        table.fill_range(rng, width_u::is_2);
+    }
+    return table;
+}
+
+// Read data from "EastAsianWidth.txt".
+// The latest version can be found at:
+// https://www.unicode.org/Public/UCD/latest/ucd/EastAsianWidth.txt
+// The current implementation works for:
+// https://www.unicode.org/Public/15.0.0/ucd/EastAsianWidth.txt
+// To make this function work, the file should not contain a BOM.
+table_u read_from(std::ifstream& source) {
+    using namespace std;
+    table_u table;
+
+    // "The unassigned code points in the following blocks default to "W":"

```

Comment:
About this comment: I'm taking the first line(dropping `All undesignated code ...`) here:
![image](https://github.com/microsoft/STL/assets/60953653/2d8fb0b0-59f2-4277-b49b-e2cc06ce0bb1)
[tr11(also in v 15.0.0)](https://www.unicode.org/reports/tr11/#:~:text=the%20Supplementary%20Ideographic,Plane%2C%2030000..3FFFF) refer to these blocks as a whole. However, I'm confused why the ranges are different here (`20000..2FFFF` instead of `20000, 2FFFD`; `30000..3FFFF` instead of `30000..3FFFD`). To get the same result as in `Annex: Exhaustive list of changes` in the paper I have to choose `20000, 2FFFD` and `30000..3FFFD`.
![image](https://github.com/microsoft/STL/assets/60953653/8155603d-96a9-4bb8-9579-f38a257d4142)

Am I missing something? Or is this an error in the tr11 document?